source/frontend/src/components/maps/leaflet/LayerPopup/LayerPopupContainer.tsx

```diff
@@ -52,37 +58,60 @@ export const LayerPopupContainer = React.forwardRef<LeafletPopup, React.PropsWit
       if (mapMachine.mapLocationFeatureDataset) {
         const featureSet = mapMachine.mapLocationFeatureDataset;
 
-        let popupTitle = 'Location Information';
-        let popupMapBounds: LatLngBounds | undefined;
-        let popupDisplayConfig: PopupContentConfig = {};
-        let popupProperties: GeoJsonProperties = {};
-        let popupFeature;
+        const layersData: LayerData[] = [];
 
         if (featureSet.parcelFeature !== null) {
-          popupTitle = 'LTSA ParcelMap data';
-          popupMapBounds = featureSet.parcelFeature.geometry
+          const parcelData: LayerData = { title: 'LTSA ParcelMap data', data: null, config: {} };
+
+          parcelData.bounds = featureSet.parcelFeature.geometry
             ? geoJSON(featureSet.parcelFeature.geometry).getBounds()
             : undefined;
-          popupDisplayConfig = parcelLayerPopupConfig;
-          popupProperties = featureSet.parcelFeature.properties;
-          popupFeature = featureSet.parcelFeature;
-        } else if (featureSet.municipalityFeature !== null) {
-          popupTitle = 'Municipality Information';
-          popupMapBounds = featureSet.municipalityFeature.geometry
+          parcelData.config = parcelLayerPopupConfig;
+          parcelData.data = featureSet.parcelFeature.properties;
+          parcelData.feature = featureSet.parcelFeature;
+          layersData.push(parcelData);
+        }
+
+        if (featureSet.municipalityFeature !== null) {
+          const parcelData: LayerData = {
+            title: 'Municipality Information',
+            data: null,
+            config: {},
+          };
+
+          parcelData.bounds = featureSet.municipalityFeature.geometry
             ? geoJSON(featureSet.municipalityFeature.geometry).getBounds()
             : undefined;
-          popupDisplayConfig = municipalityLayerPopupConfig;
-          popupProperties = featureSet.municipalityFeature.properties;
-          popupFeature = featureSet.municipalityFeature;
+          parcelData.config = municipalityLayerPopupConfig;
+          parcelData.data = featureSet.municipalityFeature.properties;
+          parcelData.feature = featureSet.municipalityFeature;
+          layersData.push(parcelData);
+        }
+
+        setLayerPopup({
+          latlng: mapMachine.mapLocationFeatureDataset.location,
+          layers: layersData,
+        });
```

Comment:
I see duplicated calls to "setLayerPopup()" here and later on line 112... 
Given that we are already calling it in line 112 - could this call be removed?